packages/api/src/__tests__/useWS.spec.tsx

```diff
@@ -116,20 +86,21 @@ describe('useWS', () => {
             ),
         });
 
-        render(<UseWSExample name={'cashier'} request={{ cashier: 'deposit' }} />);
-
-        const is_loading = screen.getByTestId('dt_is_loading');
-        const error = screen.getByTestId('dt_error');
-        const data = screen.getByTestId('dt_data');
-        const send = screen.getByTestId('dt_send');
-
-        expect(is_loading).toHaveTextContent('false');
-        expect(error).toHaveTextContent('undefined');
-        expect(data).toHaveTextContent('undefined');
-        userEvent.click(send);
-        await waitFor(() => expect(is_loading).toHaveTextContent('true'));
-        await waitFor(() => expect(data).toHaveTextContent('undefined'));
-        await waitFor(() => expect(error).toHaveTextContent('{"code":"ASK_TNC_APPROVAL","message":"Error message"}'));
-        await waitFor(() => expect(is_loading).toHaveTextContent('false'));
+        const { result, waitForNextUpdate } = renderHook(() => useWS('cashier'));
+
+        expect(result.current.is_loading).toBe(false);
+        expect(result.current.error).toBe(undefined);
+        expect(result.current.data).toBe(undefined);
+
+        act(() => {
+            result.current.send({ cashier: 'deposit' });
+        });
+        // userEvent.click(send);
+        expect(result.current.is_loading).toBe(true);
+        await waitForNextUpdate();
+        expect(result.current.data).toBe(undefined);
+        // await waitForNextUpdate();
```

Comment:
```suggestion
        expect(result.current.is_loading).toBe(true);
        await waitForNextUpdate();
        expect(result.current.data).toBe(undefined);
```
@hamid-deriv Should we remove the commented lines? 🤔

Reply:
Of course!!!